### PR TITLE
Add Pokemon general info component

### PR DESCRIPTION
**Pull request by piotr-wawro**

### CODE REVIEW

File: src/components/PokemonGeneralInfo.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import {StyleSheet, View} from 'react-native'
+import { useTheme, Text } from 'react-native-paper';
+import PokemonType from './PokemonType';
+
+export default function PokemonGeneralInfo({name, id, weight, height, type1, type2=null}) {
+  const colors = useTheme().colors;
+
```
Comment (Blazevarjo) [inline src/components/PokemonGeneralInfo.js]: Better it's just types with array of types
```suggestion

export default function PokemonGeneralInfo({name, id, weight, height, types}) {
  const colors = useTheme().colors;

```

File: src/components/PokemonGeneralInfo.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      <Text style={styles.id}>
+        {"#" + id.padStart(3, "0")}
+      </Text>
+
+      <View style={styles.typesContainer}>
+        <PokemonType type={type1} passedStyle={styles.type}/>
+        {type2 && <PokemonType type={type2} passedStyle={styles.type}/>}
+      </View>
+
```
Comment (Blazevarjo) [inline src/components/PokemonGeneralInfo.js]: that's cleaner with array approach
```suggestion

      <View style={styles.typesContainer}>
       types.map((type, index) => <PokemonType key={index} type={type} passedStyle={styles.type}/>)
      </View>

```

File: src/components/PokemonType.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ export default function PokemonType({ passedStyle, type }) {
   const typeColor = colors.pokemon.types[type];
   return (
     <View style={[styles.container, { backgroundColor: typeColor }, passedStyle]}>
-      <Text style={[styles.content, { color: colors.white }]}>{type.toUpperCase()}</Text>
+      <Text style={[styles.content, { color: colors.white }, passedStyle]}>{type.toUpperCase()}</Text>
     </View>
   );
```
Comment (Blazevarjo) [inline src/components/PokemonType.js]: I think if we want to change text component's style we should also change props to differentiate styles
```
<View style={[styles.container, { backgroundColor: typeColor }, containerStyle]}>
    <Text style={[styles.content, { color: colors.white }, textStyle]}>{type.toUpperCase()}</Text>
</View>
```

**Review by Blazevarjo — CHANGES_REQUESTED** [review]
Looks good

**Review by kamil-matula — CHANGES_REQUESTED** [review]
I agree with @Blazevarjo - try to add array of types instead of two different types.

**Review by kamil-matula — APPROVED** [review]

**Review by Blazevarjo — APPROVED** [review]

**Pull request merged**